tests/integrations/raftstore/test_replication_mode.rs

```diff
@@ -0,0 +1,179 @@
+        Err(mpsc::RecvTimeoutError::Timeout)
+    );
+    must_get_none(&cluster.get_engine(1), b"k2");
+    let state = cluster.pd_client.region_replication_status(region.get_id());
+    assert_eq!(state.state_id, 1);
+    assert_eq!(state.state, RegionReplicationState::IntegrityOverLabel);
+}
+
+/// Conf change should consider labels when DrAutoSync is chosen.
+#[test]
+fn test_check_conf_change() {
+    let mut cluster = prepare_cluster();
+    let pd_client = cluster.pd_client.clone();
+    pd_client.must_remove_peer(1, new_peer(2, 2));
+    must_get_none(&cluster.get_engine(2), b"k1");
+    cluster.add_send_filter(IsolationFilterFactory::new(2));
+    pd_client.must_add_peer(1, new_learner_peer(2, 4));
+    let region = cluster.get_region(b"k1");
+    // Peer 2 can be promoted as there will be enough quorum alive.
+    let cc = new_change_peer_request(ConfChangeType::AddNode, new_peer(2, 4));
+    let req = new_admin_request(region.get_id(), region.get_region_epoch(), cc);
+    let res = cluster
+        .call_command_on_leader(req, Duration::from_secs(3))
+        .unwrap();
+    assert!(!res.get_header().has_error(), "{:?}", res);
+    must_get_none(&cluster.get_engine(2), b"k1");
+    cluster.clear_send_filters();
+    must_get_equal(&cluster.get_engine(2), b"k1", b"v0");
+
+    pd_client.must_remove_peer(1, new_peer(3, 3));
+    must_get_none(&cluster.get_engine(3), b"k1");
+    cluster.add_send_filter(IsolationFilterFactory::new(3));
+    pd_client.must_add_peer(1, new_learner_peer(3, 5));
+    let region = cluster.get_region(b"k1");
+    // Peer 2 can be promoted as there will be enough quorum alive.
+    let cc = new_change_peer_request(ConfChangeType::AddNode, new_peer(3, 5));
+    let req = new_admin_request(region.get_id(), region.get_region_epoch(), cc);
+    let res = cluster
+        .call_command_on_leader(req, Duration::from_secs(3))
+        .unwrap();
+    assert!(
+        res.get_header()
+            .get_error()
+            .get_message()
+            .contains("unsafe to perform conf change peer"),
+        "{:?}",
+        res
+    );
+}
+
+// Tests if group id is updated when adding new node and apply snapshot.
+#[test]
+fn test_update_group_id() {
+    let mut cluster = new_server_cluster(0, 3);
+    let pd_client = cluster.pd_client.clone();
+    pd_client.disable_default_operator();
+    pd_client.configure_dr_auto_sync("zone");
+    cluster.create_engines();
+    cluster.bootstrap_conf_change();
+    cluster.cfg.raft_store.pd_store_heartbeat_tick_interval = ReadableDuration::millis(50);
+    cluster.cfg.raft_store.raft_log_gc_threshold = 10;
+    cluster
+        .cfg
+        .server
+        .labels
+        .insert("zone".to_owned(), "ES".to_owned());
+    cluster.run_node(1).unwrap();
+    cluster.run_node(2).unwrap();
+    cluster
+        .cfg
+        .server
+        .labels
+        .insert("zone".to_owned(), "WN".to_owned());
+    cluster.run_node(3).unwrap();
+    cluster.must_put(b"k1", b"v0");
```

Comment:
Why not use `prepare_cluster`?

Reply:
Need to change label value during setup.